src/app/CommandSender.cpp

```diff
@@ -412,17 +412,19 @@ CHIP_ERROR CommandSender::ProcessInvokeResponseIB(InvokeResponseIB::Parser & aIn
 
 CHIP_ERROR CommandSender::SetCommandSenderConfig(CommandSender::ConfigParameters & aConfigParams)
 {
-#if CHIP_CONFIG_SENDING_BATCH_COMMANDS_ENABLED
```

Comment:
So this was not a "reduce codesize" kind of ifdef?

Reply:
Right now it doesn't seem to be doing much based on the size comparison report. With the exception for on target on `bl702l`, which seems like read and write section increased by 500 bytes, while having a decrease in code size (text).

We could be more aggressive in using this ifdef to reduce code usage. Although historically this was initially added in case there was anything important to block on this, but with the latest addition of `CommandSender::ExtendableCallback`, which is something we want people to move to regardless of batch commands, the needs for `CHIP_CONFIG_SENDING_BATCH_COMMANDS_ENABLED` seems to be diminishing.

With that all that said, Tennessee did mention that `CommandSender` should be doing more than it is today for tracking which commands have come back. So I will hold off on merging this PR until that lands and we can see if there is a code size reduction worth keeping around

Reply:
Additionally one reason that I wanted to remove this `CHIP_CONFIG_SENDING_BATCH_COMMANDS_ENABLED` was to remove some test gating on cleaning up some unit tests to use CommandSender to build batch commands

https://github.com/project-chip/connectedhomeip/pull/31081